### PR TITLE
feat: Add Bitbucket provider

### PR DESCRIPTION
**Pull request by svenax**
Add Bitbucket as a standard provider. While less common than GitHub/GitLab, it is used by a fair amount of people.

### CODE REVIEW

**Review by pawamoy — APPROVED** [review]
Thanks!

And thanks again for the typo fixes :slightly_smiling_face:

File: config/ruff.toml [hunk 1]
```diff
@@ -71,6 +71,7 @@ ignore = [
     "PLR0915",  # Too many statements
     "SLF001", # Private member accessed
     "TRY003",  # Avoid specifying long messages outside the exception class
+    "UP032",  # Use f-string instead of `format` call
```
[the remark this answers is not in the document]
Comment (pawamoy) [reply config/ruff.toml:74]: Oh OK `.format` is used elsewhere as well :+1:

**Comment by pawamoy** [comment]
Can you resolve conflicts please? Then we can merge and release :slightly_smiling_face:

File: config/ruff.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (svenax) [reply config/ruff.toml:74]: Used in quite a few places, so I thought it was ugly to annotate all of them.

**Comment by svenax** [comment]
Python 3.12 has a hard deprecation for `datetime.utcfromtimestamp()`. Do you want me to fix that too?

**Comment by pawamoy** [comment]
If you have the capacity, yes please, otherwise don't worry about it and I'll fix it later.

**Comment by svenax** [comment]
No problem, I'll fix that.

File: src/git_changelog/cli.py [hunk 2]
```diff
@@ -358,7 +358,6 @@ def build_and_render(
         with open(output, "w") as changelog_file:  # type: ignore[arg-type]
             changelog_file.write("\n".join(lines).rstrip("\n") + "\n")
 
-    # overwrite output file
```
Comment (pawamoy) [inline src/git_changelog/cli.py:361]: Why removing this comment?

Comment (svenax) [reply src/git_changelog/cli.py:361]: Hmm, that was not intentional. Probably it happened merging your updates. I'll put it back.

**Review by pawamoy — APPROVED** [review]
Awesome work, thanks a lot!

**Comment by pawamoy** [comment]
Released in v2.1.0 :slightly_smiling_face:

**Pull request merged**